src/services/item/plugins/embeddedLink/service.ts

```diff
@@ -0,0 +1,78 @@
+import fetch from 'node-fetch';
+
+import { FastifyBaseLogger } from 'fastify';
+
+type IframelyLink = {
+  rel: string[];
+  href: string;
+};
+
+type IframelyResponse = {
+  meta: {
+    title?: string;
+    description?: string;
+  };
+  html: string;
+  links: IframelyLink[];
+};
+
+const hasRel = (rel: string[], value: string) => rel.some((r) => r === value);
+const hasThumbnailRel = (rel: string[]) => hasRel(rel, 'thumbnail');
+const hasIconRel = (rel: string[]) => hasRel(rel, 'icon');
+
+const CSP_KEY = 'Content-Security-Policy';
+const X_FRAME_KEY = 'X-Frame-Options';
+const CSP_FRAME_NONE = ["frame-ancestors 'none'", "frame-ancestors 'self'"];
+const X_FRAME_DISABLED = ['sameorigin', 'deny'];
+
+export class EmbeddedLinkService {
+  async getLinkMetadata(iframelyHrefOrigin: string, url: string) {
+    const response = await fetch(`${iframelyHrefOrigin}/iframely?uri=${encodeURIComponent(url)}`);
+    // better clues on how to extract the metadata here: https://iframely.com/docs/links
+    const { meta = {}, html, links = [] } = (await response.json()) as IframelyResponse;
+    const { title, description } = meta;
+
+    return {
+      title: title?.trim(),
+      description: description?.trim(),
+      html,
+      thumbnails: links.filter(({ rel }) => hasThumbnailRel(rel)).map(({ href }) => href),
+      icons: links.filter(({ rel }: { rel: string[] }) => hasIconRel(rel)).map(({ href }) => href),
+    };
+  }
+
+  async checkEmbeddingAllowed(url: string, logger?: FastifyBaseLogger): Promise<boolean> {
+    try {
+      const { headers } = await fetch(url);
```

Comment:
Does it work if you fetch with a `HEAD` request instead ? This would reduce the bandwidth of the page.

Reply:
I tried but it doesn't seems to work with the HEAD request.

Reply:
okay !